anvill/src/Providers/MemoryProvider.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021 Trail of Bits, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <anvill/Providers/MemoryProvider.h>
+
+#include <anvill/Program.h>
+
+namespace anvill {
+namespace {
+
+// Provider of memory wrapping around an `anvill::Program`.
+class ProgramMemoryProvider final : public MemoryProvider {
+ public:
+
+  virtual ~ProgramMemoryProvider(void) = default;
```

Comment:
A `final` class with a virtual default destructor? As above, is this needed?

Reply:
No, typed it out. I don't use IDE templates / code generation.